cabal-install/src/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -1568,7 +1568,7 @@ legacyPackageConfigFieldDescrs =
             dispFlagAssignment
             parsecFlagAssignment
             configConfigurationsFlags
-            (\v conf -> conf{configConfigurationsFlags = v})
+            (\v conf -> conf{configConfigurationsFlags = configConfigurationsFlags conf <> v})
```

Comment:
Perhaps this indicates there should be a helper like `commaListField` which operates on a generic `Monoid` rather than a list. Otherwise, looks like a precise fix to me :+1:

Reply:
Good idea, I've added one.